Board.py

```diff
@@ -1,11 +1,163 @@
+    return abs(end_x - start_x) > 1
 
 
 class Board:
+
     def __init__(self):
-        rows, cols = 8, 8
-        self.board = [['0' for _ in range(cols)] for _ in range(rows)]
+        self.player_one_pieces = []
+        self.player_two_pieces = []
+        for i in range(3):
+            for z in range(4):
+                checker_x = i
+                checker_y = z * 2 + (i + 1) % 2
+                temp_piece = self.generate_checker(checker_x, checker_y, Player.player1)
+                self.player_one_pieces.append(temp_piece)
+
+                checker_x = i + 5
+                checker_y = z * 2 + i % 2
+                temp_piece = self.generate_checker(checker_x, checker_y, Player.player2)
+                self.player_two_pieces.append(temp_piece)
+
+    def move(self, move):  # move tuple ==> (start_x, start_y, end_x, end_y, Player)
+        start_x = move[0]
+        start_y = move[1]
+        end_x = move[2]
+        end_y = move[3]
+        player = move[4]
+
+        if is_jump(start_x, end_x):
+            self.remove_checker(int((start_x + end_x) / 2), int((start_y + end_y) / 2))  # remove piece
+
+        self.remove_checker(start_x, start_y)
+        new_checker = self.generate_checker(end_x, end_y, player)
+        self.add_new_checker(new_checker)
+
+    def generate_checker(self, x, y, player): # checker structure ==> (x, y, Player, Piece)
+        if y == 0 or y == ROWS:
+            return x, y, player, Piece.king
+        else:
+            return x, y, player, Piece.basic
+
+    def add_new_checker(self, checker):
+        if checker[3] == Player.player1:
+            self.player_one_pieces.append(checker)
+        elif checker[3] == Player.player2:
+            self.player_two_pieces.append(checker)
 
     def print_board(self):
-        for row in self.board:
-            print(' '.join(row))
+        indicator = 0
+        board = self.generate_board()
+        for my_row in board:
+            print(indicator, end="")
+            print(' ', end="")
+            print(' '.join(my_row))
+            indicator = indicator + 1
+        print('  ', end="")
+        for i in range(8):
+            print(i, end=' ')
+        print()
+
+    def get_checker(self, x, y):
+        for checker in self.player_one_pieces:
+            if checker[0] == x and checker[1] == y:
+                return checker
+        for checker in self.player_two_pieces:
+            if checker[0] == x and checker[1] == y:
+                return checker
+        return -1  # checker not found at given position
+
+    def remove_checker(self, x, y):
+        checker = self.get_checker(x, y)
+        if checker[2] == Player.player1:
+            self.player_one_pieces.remove(checker)
+        elif checker[2] == Player.player2:
+            self.player_two_pieces.remove(checker)
+        print("something terribly wrong has happened!")  # consider gracefully handling error here
+        return 0
+
+    def get_available_moves(self, checker):
```

Comment:
theres cleaner ways to implement this function but i'm getting tired and lazy so address the above comments first